src/execution_plan/ops/op_conditional_traverse.c

```diff
@@ -76,14 +76,23 @@ OpBase *NewCondTraverseOp(const ExecutionPlan *plan, Graph *g, AlgebraicExpressi
 	op->recordsCap = 0;
 	op->recordCount = 0;
 	op->edge_ctx = NULL;
+	op->dest_label = NULL;
+	op->dest_label_id = GRAPH_NO_LABEL;
 
 	// Set our Op operations
 	OpBase_Init((OpBase *)op, OPType_CONDITIONAL_TRAVERSE, "Conditional Traverse", CondTraverseInit,
 				CondTraverseConsume, CondTraverseReset, CondTraverseToString, CondTraverseClone, CondTraverseFree,
 				false, plan);
 
 	assert(OpBase_Aware((OpBase *)op, AlgebraicExpression_Source(ae), &op->srcNodeIdx));
-	op->destNodeIdx = OpBase_Modifies((OpBase *)op, AlgebraicExpression_Destination(ae));
+	const char *dest = AlgebraicExpression_Destination(ae);
+	op->destNodeIdx = OpBase_Modifies((OpBase *)op, dest);
+	// Check the QueryGraph node and retrieve label data if possible.
+	QGNode *dest_node = QueryGraph_GetNodeByAlias(plan->query_graph, dest);
+	if(dest_node->labelID != GRAPH_NO_LABEL) {
```

Comment:
It seems like `QGNode_New` can create a `QGNode` object where the label is set but the `labelID` is `GRAPH_NO_LABEL` a contradicting state, which might lead to this condition to fail although `label` is set to an actual existing label.

Reply:
`QGNode_New` only has one caller (`_BuildQueryGraphAddNode`), which always passes NULL as the `label` value. I will rewrite its signature to remove `label`.